common/CommonAPI.h

```diff
@@ -1566,10 +1485,17 @@ class CommonAPI
 		const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device,
 		nbl::core::smart_refctd_ptr<nbl::video::ISwapchain> swapchain,
 		nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> renderpass, 
-		nbl::asset::E_FORMAT depthFormat = nbl::asset::EF_UNKNOWN)->std::array<nbl::core::smart_refctd_ptr<nbl::video::IGPUFramebuffer>, InitOutput::MaxSwapChainImageCount>
+		nbl::asset::E_FORMAT baseDepthFormat = nbl::asset::EF_UNKNOWN
+	) -> std::array<nbl::core::smart_refctd_ptr<nbl::video::IGPUFramebuffer>, InitOutput::MaxSwapChainImageCount>
 	{
 		using namespace nbl;
-		bool useDepth = asset::isDepthOrStencilFormat(depthFormat);
+
+		auto depthFormat = device->getPhysicalDevice()->promoteImageFormat(
+			{ baseDepthFormat, nbl::video::IPhysicalDevice::SFormatImageUsage(nbl::asset::IImage::EUF_DEPTH_STENCIL_ATTACHMENT_BIT) },
+			nbl::asset::IImage::ET_OPTIMAL
+		);
+		bool useDepth = depthFormat != nbl::asset::EF_UNKNOWN;
```

Comment:
figure this out before promotion, we don't want to drop the usage of a depth buffer just because someone couldn't promote